src/plugins/aws/idc/index.ts

```diff
@@ -0,0 +1,194 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { validateResponse } from "../../../common/fetch";
+import { retryWithSleep } from "../../../common/retry";
+import { cached } from "../../../drivers/auth";
+import {
+  AWSAuthorizeResponse,
+  AWSClientInformation,
+  AWSTokenResponse,
+} from "../../../types/aws/oidc";
+import { OidcLoginStepHelpers } from "../../../types/oidc";
+import { DEVICE_GRANT_TYPE, oidcLogin } from "../../oidc/login";
+import { AwsCredentials } from "../types";
+
+const AWS_TOKEN_EXPIRY = 3600e3;
```

Comment:
suggestion: add time constants, and use
```suggestion
const AWS_TOKEN_EXPIRY = 1 * HOURS;
```
here